app/assets/javascripts/angular/controllers/dialog_controller.js

```diff
@@ -0,0 +1,61 @@
+(function(){
+    angular
+    .module('taxi')
+    .controller('DialogController', ['$scope', '$http', function($scope, $http) {
+
+        var dispatcher = new WebSocketRails(window.location.host + '/websocket');
+
+        $scope.options = {
+            availableOptions: [
+            {value: '1'}, {value: '2'}, {value: '3'}, {value: '4'}, 
+            {value: '5'}, {value: '6'}, {value: '7'}, {value: '8'}
+            ],
+            selectedOption: {value: '1'}
+        };
+        $scope.phone_pattern = /(0)[0-9]{9}/;
+        $scope.email_pattern = /^(.+)@(.+)$/;
+        $scope.isDispatcher = true;
+
+        function checkStatus(data){
+            if (data.driver_id) {
+                data.driver_id = data.driver_id.id;
+                data.status = 'waiting';
+            }
+        }
+        
+        function putMethod(data) {
```

Comment:
Don't call functions 'putMethod', function name should reflect function purpose.
Also create a service for such functions (working with API) and move all them there.